File: Java/ex3.java
class ContaBancaria {
    private double saldo;
    public String titular;

    public ContaBancaria(String titular, double saldoInicial) {
        this.saldo = saldoInicial;
        this.titular = titular;
    }

    public String depositar(double valor) {
        if (valor > 0) {
            saldo += valor;
            return String.format("Depósito de R$%.2f realizado. Saldo atual: R$%.2f", valor, saldo);
        } else {
            return "Valor de depósito inválido!";
        }
    }

    public String sacar(double valor) {
        if (valor > 0 && valor <= saldo) {
            saldo -= valor;
            return String.format("Saque de R$%.2f realizado. Saldo atual: R$%.2f", valor, saldo);
        } else {
            return "Saldo insuficiente ou valor de saque inválido!";
        }
    }

    public String exibirSaldo() {
        return String.format("Saldo atual de %s: R$%.2f", titular, saldo);
    }
}

public class Main {
    public static void main(String[] args) {
        ContaBancaria contaTeste = new ContaBancaria("João Silva", 1000);
        System.out.println(contaTeste.depositar(500));
        System.out.println(contaTeste.sacar(300));
        System.out.println(contaTeste.exibirSaldo());
    }
}

File: Python/ex3.py
class ContaBancaria:
    def __init__(self, titular, saldo_inicial=0):
        self.__saldo = saldo_inicial  # Atributo saldo encapsulado (privado)
        self.titular = titular

    def depositar(self, valor):
        if valor > 0:
            self.__saldo += valor
            return f"Depósito de R${valor:.2f} realizado. Saldo atual: R${self.__saldo:.2f}"
        else:
            return "Valor de depósito inválido!"

    def sacar(self, valor):
        if 0 < valor <= self.__saldo:
            self.__saldo -= valor
            return f"Saque de R${valor:.2f} realizado. Saldo atual: R${self.__saldo:.2f}"
        else:
            return "Saldo insuficiente ou valor de saque inválido!"

    def exibir_saldo(self):
        return f"Saldo atual de {self.titular}: R${self.__saldo:.2f}"


conta_teste = ContaBancaria("João Silva", 1000)

print(conta_teste.depositar(500))
print(conta_teste.sacar(300))
print(conta_teste.exibir_saldo())
